Source/PersonalityInsightsV3/PersonalityInsights.swift

```diff
@@ -0,0 +1,274 @@
+            rawScores: rawScores,
+            consumptionPreferences: consumptionPreferences,
+            failure: failure,
+            success: success
+        )
+    }
+
+    /**
+     Analyze content to generate a personality profile.
+ 
+     - parameter version: The date in the form "YYYY-MM-DD" that will specify a version no later than the date provided.
+     - parameter content: The content to analyze.
+     - parameter contentType: The MIME content-type of the content.
+     - parameter acceptLanguage: The desired language of the response.
+     - parameter contentLanguage: The language of the text being analyzed.
+     - parameter rawScores: If true, then a raw score for each characteristic is returned in
+        addition to a normalized score. Raw scores are not compared with a sample population.
+        A raw sampling error for each characteristic is also returned.
+     - parameter consumptionPreferences: If true, then information inferred about consumption preferences is returned in addition to the results.
+     - parameter failure: A function executed if an error occurs.
+     - parameter success: A function executed with the personality profile.
+     */
+    private func getProfile(
+        version: String,
+        fromContent content: Data?,
+        withType contentType: String,
+        acceptLanguage: String? = nil,
+        contentLanguage: String? = nil,
+        rawScores: Bool? = nil,
+        consumptionPreferences: Bool? = nil,
+        failure: ((Error) -> Void)? = nil,
+        success: @escaping (Profile) -> Void)
+    {
+        // construct query parameters
+        var queryParameters = [URLQueryItem]()
+        if let rawScores = rawScores {
+            let queryParameter = URLQueryItem(name: "raw_scores", value: "\(rawScores)")
+            queryParameters.append(queryParameter)
+        }
+        if let consumptionPreferences = consumptionPreferences {
+            let consumptionPreferenceQueryParameter = URLQueryItem(name: "consumption_preferences", value: "\(consumptionPreferences)")
+            queryParameters.append(consumptionPreferenceQueryParameter)
+        }
+        let versionQueryParameter = URLQueryItem(name: "version", value: version)
+        queryParameters.append(versionQueryParameter)
+
+        // construct header parameters
+        var headerParameters = defaultHeaders
+        if let acceptLanguage = acceptLanguage {
+            headerParameters["Accept-Language"] = acceptLanguage
+        }
+        if let contentLanguage = contentLanguage {
+            headerParameters["Content-Language"] = contentLanguage
+        }
+
+        // construct REST request
+        let request = RestRequest(
+            method: "POST",
+            url: serviceURL + "/v3/profile",
+            credentials: credentials,
+            headerParameters: headerParameters,
+            acceptType: "application/json",
+            contentType: contentType,
```

Comment:
While reviewing this code, I realized how strange it is to set some header parameters in the `headerParameters` dictionary and pass others through the `RestRequest` initializer (i.e. the `acceptType` and `contentType` in particular).

What would you think about removing the `acceptType` and `contentType` parameters from the `RestRequest` initializer, and instead requiring service classes to specify them in the `headerParameters` dictionary? I think it might be easier to understand, since all of the header parameters would be set in one place.

No need to make any changes here. I'm just curious what you would think of that change.